great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -284,3 +292,28 @@ def run(
                 )
 
         return expectation_suite
+
+    def self_check(self, pretty_print=True) -> dict:
+        """
```

Comment:
I don't love this - `test_yaml_config` assumes the resulting instantiated object has a `self_check` method. Without getting overly complicated with something like multiple inheritance, how do we ensure that each supported class in `test_yaml_config` has this method?

Reply:
Agreed - inheriting also from an abstract `TestYamlConfigEnabledClass(ABC)` which has an @abstractmethod `self_check()` would work but not be great for maintainability. We can write a test which checks that all classes in `BaseDataContext.ALL_TEST_YAML_CONFIG_SUPPORTED_TYPES` have a `self_check` method, that would at least catch issues early.

Reply:
Added a test to cover this functionality - not the cleanest implementation but it will definitely highlight issues (like it did here)